### PR TITLE
login now supports mobile number username

### PR DESCRIPTION
**Pull request by Calrider**
login now supports mobile number username
get_account_id now returns account id instead of boolean

### CODE REVIEW

File: webull.py [hunk 1]
```diff
@@ -148,12 +164,12 @@ def get_account_id(self):
...
         response = requests.get(self.urls.account_id(), headers=headers)
         result = response.json()
-
         if result['success']:
-            self.account_id = str(result['data'][0]['secAccountId'])
-            return True
+            id = str(result['data'][0]['secAccountId'])
+            self.account_id = id
+            return id
         else:
```
[the remark this answers is not in the document]
Comment (Calrider) [reply webull.py:170]: This PR needs to be held off, I found a bug with dep import.

File: webull.py [hunk 2]
```diff
@@ -28,9 +29,9 @@ def __init__(self, cmd=False):
         self.uuid = ''
         self.did = self._get_did()
 
-        if cmd == True :
+        if cmd == True:
```
[the remark this answers is not in the document]
Comment (Calrider) [reply webull.py]: yes. there are a lot of things like this throughout the code base. will refactor as I find them.

File: webull.py [hunk 3]
```diff
@@ -805,8 +820,9 @@ def get_account_id(self):
         response = requests.get(self.urls.paper_account_id(),
                                 headers=headers)
         result = response.json()
-        self.paper_account_id = result[0]['id']
-        return True
+        id = result[0]['id']
+        self.paper_account_id = id
+        return id
```
[the remark this answers is not in the document]
Comment (Calrider) [reply webull.py:825]: yup

File: webull.py [hunk 4]
```diff
@@ -65,27 +66,42 @@ def build_req_headers(self, include_trade_token=False, include_time=False):
...
-                'accountType': 2,
-                'deviceId': self.did,
-                'pwd': md5_hash.hexdigest(),
-                'regionId': 1}
-        response = requests.post(self.urls.login(), json=data, headers=self.headers)
 
+        try:
+          validate_email(username)
+          accountType = 2 # email
+        except EmailNotValidError as e:
```
[the remark this answers is not in the document]
Comment (Calrider) [reply webull.py:85]: Was planning to write `e` to a logger. Any preference of logger or just stdlib `logging`?

[the remark this answers is not in the document]
Comment (gamesover) [reply webull.py:85]: ah, just choose what you prefer :) I did not use logger for python before

**Comment by Calrider** [comment]
Will be closing this PR as I have squashed a bunch of recent commits together. Thx for the feedback

**Pull request closed without merging**